hw_4/script.js

```diff
@@ -0,0 +1,25 @@
+function factorial(n) {
+  return n <= 1 ? 1 : n * factorial(n - 1);
+}
+
+var arr1 = [];
+
+function heapFilling() {
+  for (let i = 0; i < 1024 * 1024 * 108; i++) {
+    arr1.push(1);
+    // If you uncomment the next line of code and run in node.js, no heap overflow occurs...
+    // I think when console.log is called, the engine optimizations have time to work.
+    // console.log(process.memoryUsage().heapUsed);
+  }
+}
+
+// Caution! Calling the next part of the code will result in a stack overflow.
```

Comment:
Use single сomment for description and a multi-line сomment for the code we can use, so it is clearer.
```suggestion
// Caution! Calling the next part of the code will result in a stack overflow.

/*try {
  console.log("Recursive function factorial of a 100000 will cause a stack overflow: ");
  console.log(factorial(100000));
} catch (e) {
  console.error(e.message);
}*/
```

Reply:
https://github.com/BlackTruth/javascript/blob/4d02cb8e9b7adf484665c236844a5d36591978a6/hw_4/script.js#L17-L27
Fixed.